docs/language-server/protocol-language-server.md

```diff
@@ -367,7 +367,7 @@ interface ExpressionUpdate {
   /** The id of updated expression. */
   expressionId: ExpressionId;
   /** The updated type of the expression. */
```

Comment:
The current docs are not clear about some details of the protocol. An empty array seems to be treated specially; judging from the implementation, it looks like:
- When the status is `Pending`, an empty array can indicate that the type has not been determined yet.
- When the status is not `Pending`, an empty array indicates that no specific type is known for the value--maybe this can happen either if the type is known to be `Any`. Can this also happen if the type could not be determined?

Are these updates incremental? It's not clear whether in the pending state `[]` might be sent to indicate "no change", or every update will contain all currently-known type information even if it hasn't changed.

Reply:
I updated the docs. The logic did not change. The array is just a way to encode intersection types
- `[]` indicates no type information (previously `null`)
- `[T]` represent a type of the expression (previously `T`)
- `[T,U]` represents an intersection type `T & U` (previously not supported, probably delivered just as a `T` type)